restapi/admin_users.go

```diff
@@ -467,3 +475,65 @@ func getAddUsersListToGroupsResponse(session *models.Principal, params admin_api
 
 	return nil
 }
+
+func getListUsersWithAccessToBucketResponse(session *models.Principal, bucket string) ([]string, *models.Error) {
+	ctx := context.Background()
+	mAdmin, err := newMAdminClient(session)
+	if err != nil {
+		return nil, prepareError(err)
+	}
+	// create a minioClient interface implementation
+	// defining the client to be used
+	adminClient := adminClient{client: mAdmin}
+
+	users, err := listUsers(ctx, adminClient)
+	if err != nil {
+		return nil, prepareError(err)
+	}
+	var retval []string
+	seen := make(map[string]bool)
+	for i := 0; i < len(users); i++ {
```

Comment:
This is just a suggestion, to iterate the array of objects you can also do:
```suggestion
	for _, user := range users {
```

Inside the loop you can just use the `user` object directly, ie : `user.AccessKey` instead of `users[i].AccessKey`